crates/toml/src/de.rs

```diff
@@ -38,9 +38,9 @@
 /// assert_eq!(config.owner.name, "Lisa");
 /// ```
 #[cfg(feature = "parse")]
-pub fn from_str<T>(s: &'_ str) -> Result<T, Error>
+pub fn from_str<'a, T>(s: &'a str) -> Result<T, Error>
 where
-    T: serde::de::DeserializeOwned,
+    T: serde::de::Deserialize<'a>,
```

Comment:
> This change should be semver-compatible according to my best interpretation of the [API evolution RFC](https://github.com/rust-lang/rfcs/blob/master/text/1105-api-evolution.md).

This changes the trait bounds from a more general one to a more specific one which is a breaking change

Reply:
Thanks for the review! :smile: Isn't it the reverse, though? [`DeserializeOwned` is syntax sugar for the higher-ranked trait bound `for<'de> Deserialize<'de>`](https://docs.rs/serde/1.0.152/serde/de/trait.DeserializeOwned.html), so I'd argue that `DeserializeOwned` is more specific. The [HRTB's RFC seems to support this](https://rust-lang.github.io/rfcs/0387-higher-ranked-trait-bounds.html#subtyping-of-trait-references), stating that:

> [...] if I have a trait reference like `for<'a> FnMut(&'a int)`, that would be usable wherever a trait reference with a concrete lifetime, like `FnMut(&'static int)`, is expected.